pkg/resource/predicates.go

```diff
@@ -38,32 +41,49 @@ func NewPredicates(fn PredicateFn) predicate.Funcs {
 }
 
 // HasClassReferenceKind accepts ResourceClaims that reference the correct kind of resourceclass
-func HasClassReferenceKind(k ClassKind) PredicateFn {
+func HasClassReferenceKind(m manager.Manager, k ClassKinds) PredicateFn {
```

Comment:
I'd prefer to have this function take a `client.Client` and a `runtime.ObjectCreater` rather than `manager.Manager`, which is a pretty huge interface that we do not use much of.